src/MiniProfiler.Providers.StackExchange.Redis/RedisStorage.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using StackExchange.Profiling.Helpers;
+using StackExchange.Redis;
+
+namespace StackExchange.Profiling.Storage
+{
+    /// <summary>
+    /// StackExchange.Redis based storage provider for <see cref="MiniProfiler"/> results.
+    /// </summary>
+    public class RedisStorage : IAsyncStorage
+    {
+        private readonly IDatabase _database;
+
+        /// <summary>
+        /// Gets or sets the key prefix for individual profiling results. Default is "MiniProfiler_Result_".
+        /// </summary>
+        public RedisKey ProfilerResultKeyPrefix { get; set; } = "MiniProfiler_Result_";
+
+        /// <summary>
+        /// Gets or sets the list key for the profiling results list. Default is "MiniProfiler_ResultList".
+        /// </summary>
+        public RedisKey ProfilerResultListKey { get; set; } = "MiniProfiler_ResultList";
+
+        /// <summary>
+        /// Gets or sets the key prefix for the per-user set of unviewed profiling results. Default is "MiniProfiler_UnviewedResultSet_".
+        /// </summary>
+        public RedisKey ProfilerResultUnviewedSetKeyPrefix { get; set; } = "MiniProfiler_UnviewedResultSet_";
+
+        /// <summary>
+        /// Gets or sets how long to cache each <see cref="MiniProfiler"/> for, in absolute terms. Default is 60 minutes.
+        /// </summary>
+        public TimeSpan CacheDuration { get; set; } = TimeSpan.FromMinutes(60);
+
+        /// <summary>
+        /// Gets or sets the maximum number of profiling results that will be stored in the profiling list. Default is 100.
+        /// </summary>
+        public int ResultListMaxLength { get; set; } = 100;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RedisStorage"/> class with the specified Redis database.
+        /// </summary>
+        /// <param name="database">The Redis database to use.</param>
+        public RedisStorage(IDatabase database)
+        {
+            _database = database ?? throw new ArgumentNullException(nameof(database));
+        }
+
+        /// <summary>
+        /// List the latest profiling results.
+        /// </summary>
+        /// <param name="maxResults">The maximum number of results to return.</param>
+        /// <param name="start">(Optional) The start of the date range to fetch.</param>
+        /// <param name="finish">(Optional) The end of the date range to fetch.</param>
+        /// <param name="orderBy">(Optional) The order to fetch profiler IDs in.</param>
+        public IEnumerable<Guid> List(
+            int maxResults,
+            DateTime? start = null,
+            DateTime? finish = null,
+            ListResultsOrder orderBy = ListResultsOrder.Descending)
+        {
+            // TODO: start/finish not yet supported - will simply return all results
```

Comment:
I did consider using a sorted set, instead of a list, and using the prolifer's timestamp as the score. Then I can query with ZRANGEBYSCORE. But I was ever so slightly worried that write performance would suffer, especially since List is comparatively less used (this is just a bold assumption, I don't know how others use it).

But maybe the performance of sorted sets isn't worth worrying about. Thoughts?